File: ai/agent/prompt_builder.py
"""System prompt builder — assembles the context window for the AI agent."""


def build_system_prompt(
    project_context: str | None = None,
    skill_contents: list[str] | None = None,
    date: str | None = None,
) -> str:
    parts = [
        "Eres GeoAgents, un asistente de inteligencia artificial experto en análisis geoespacial, datos técnicos y automatización de tareas GIS."
    ]

    if date is None:
        from datetime import datetime
        date = datetime.now().strftime("%Y-%m-%d")

    parts.append(f"\nFecha actual: {date}")

    if project_context:
        parts.append(f"\n## Contexto del proyecto\n{project_context}")

    if skill_contents:
        parts.append(f"\n## Skills activos\n" + "\n\n---\n\n".join(skill_contents))

    return "\n".join(parts)

File: src/agent/prompt-builder.ts
/** System prompt builder (inspired by Hermes prompt_builder/). */

export interface BuildPromptOptions {
  projectContext?: string
  skillContents?: string[]
  recentMessages?: { role: string; content: string }[]
  webSearchResults?: string
  date?: string
}

const BASE_SYSTEM_PROMPT = `Eres GeoAgents, un asistente de inteligencia artificial experto en análisis geoespacial, datos técnicos y automatización de tareas GIS.`

export function buildSystemPrompt(opts: BuildPromptOptions = {}): string {
  const parts: string[] = [BASE_SYSTEM_PROMPT]

  const date = opts.date ?? new Date().toISOString().split("T")[0]
  parts.push(`\nFecha actual: ${date}`)

  if (opts.projectContext) {
    parts.push(`\n## Contexto del proyecto\n${opts.projectContext}`)
  }

  if (opts.skillContents?.length) {
    parts.push(`\n## Skills activos\n${opts.skillContents.join("\n\n---\n\n")}`)
  }

  if (opts.webSearchResults) {
    parts.push(`\n## Resultados de búsqueda web\n${opts.webSearchResults}`)
  }

  return parts.join("\n")
}

export function buildUserPrompt(content: string, _opts?: { attachments?: { name: string; type: string }[] }): string {
  return content
}
